packages/wallets/glow/src/adapter.ts

```diff
@@ -0,0 +1,270 @@
+        return this._readyState;
+    }
+
+    async connect(): Promise<void> {
+        try {
+            if (this.connected || this.connecting) {
+                return;
+            }
+            if (this._readyState !== WalletReadyState.Installed) {
+                throw new WalletNotReadyError();
+            }
+
+            this._connecting = true;
+
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const wallet = window!.solana!;
+
+            try {
+                await wallet.connect()
+            } catch (error: any) {
+                if (error instanceof WalletError) {
+                    throw error;
+                }
+                throw new WalletConnectionError(error?.message, error);
+            }
+
+            if (!wallet.publicKey) {
+                throw new WalletAccountError();
+            }
+
+            let publicKey: PublicKey;
+            try {
+                publicKey = new PublicKey(wallet.publicKey.toBytes());
+            } catch (error: any) {
+                throw new WalletPublicKeyError(error?.message, error);
+            }
+
+            wallet.on('disconnect', this._disconnected);
+
+            this._wallet = wallet;
+            this._publicKey = publicKey;
+
+            this.emit('connect', publicKey);
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        } finally {
+            this._connecting = false;
+        }
+    }
+
+    async disconnect(): Promise<void> {
+        const wallet = this._wallet;
+        if (wallet) {
+            wallet.off('disconnect', this._disconnected);
+
+            this._wallet = null;
+            this._publicKey = null;
+
+            try {
+                await wallet.disconnect();
+            } catch (error: any) {
+                this.emit('error', new WalletDisconnectionError(error?.message, error));
+            }
+        }
+
+        this.emit('disconnect');
+    }
+
+    async sendTransaction(
+        transaction: Transaction,
+        connection: Connection,
+        options?: SendTransactionOptions
+    ): Promise<TransactionSignature> {
+        try {
+            const wallet = this._wallet;
+            // Glow doesn't handle partial signers, so if they are provided, don't use `signAndSendTransaction`
+            if (wallet && 'signAndSendTransaction' in wallet && !options?.signers) {
+                // TODO: update glow to fix this
+                // HACK: Glow's `signAndSendTransaction` should always set these, but doesn't yet
+                transaction.feePayer = transaction.feePayer || this.publicKey || undefined;
+                transaction.recentBlockhash =
+                    transaction.recentBlockhash || (await connection.getRecentBlockhash('finalized')).blockhash;
+
+                const { signature } = await wallet.signAndSendTransaction(transaction, options);
+                return signature;
+            }
```

Comment:
This is undesired behavior and looks like it was copied from Phantom. It would be ideal to fix rather than duplicate.

Reply:
Yea, I'm a little confused by what's going on here haha. I don't think we need any of this code but I'm a little worried that some client that's set up to use Phantom will have a harder time integrating Glow if we leave this out.